index.rst

```diff
@@ -352,28 +385,145 @@ To make sure the system would work properly we had to enable two additional feat
+The system architecture specify mainly two types components; Controllers and Remotes, in SalObj nomenclature.
+Controllers are the base components upon which CSCs are created.
+They mainly consist of applications that read commands and write telemetry, events and acknowledgements.
+At the same time, to communicate with a CSC one would use a Remote, which write commands and read telemetry, events and acknowledgements.
+
+By allowing a separate partition for read and write operations, components can share most of the resources needed to communicate in both directions while still minimizing data duplication (as shown in :numref:`fig-partition`).
+One could argue that there could be a partition for each individual topic of the system, which would allow readers to only receive the data they are actually requesting.
+The reason for avoiding such thin partitioning is that it would require substantial resource duplication.
+For instance, DDS allows applications to share substantial resources when creating readers and writers on the same partition.
+At the same time, the software architecture allows us to safely share most of those resources for an individual component so, splitting the topics for a component into more partitions means there would be less shared resources.
+
+Details on how this is implemented can be found in `SalObj documentation <https://ts-salobj.lsst.io>`__.
+
+.. _Quality-of-Service-configuration:
+
+Quality of Service configuration
+--------------------------------
+
+The Quality of Service (QoS) are topic specify configurations that control many aspects of how information is handled by DDS, including how reliable messages are exchanged and if a topic has historical data (for later joiners).
+Our architecture apply QoS configuration based on the message type, e.g., commands, events, telemetry and acknowledgements.
+
+Up to SAL v4 and SalObj v5 the QoS was partially stored in a `QoS file <https://raw.githubusercontent.com/lsst-ts/ts_idl/v1.4.0/qos/DDS_DefaultQoS_All.xml>`__ and partially hard coded into the software code base.
+This created some issues, specially when changes needed to be done on the QoS, which required full rebuild of the system.
+As of SAL v5 and SalObj v6 this settings will be fully loaded from a QoS file stored in the `ts_idl repository <https://github.com/lsst-ts/ts_idl>`__.
+
+As part of this exercise we adopted a couple changes in the previous QoS settings.
+The most important change to be noted is that now Telemetry topics have durability set to ``VOLATILE``, which means the system will no longer store historical data.
+Since telemetry is continuously published at a set frequency by CSCs, storing historical data gives little to no benefit while imposes an additional load on the DDS systems.
+
+From now on, events are the only set of topics that are configured with ``TRANSIENT`` durability, which allows them to retain historical information.
```

Comment:
Please consider stressing that events are the only kind of topic that offers historical data. That is more important than TRANSIENT.

Perhaps something along the lines of:

"From now on, events are the only kind of topic that offer historical data. They are configured with ``TRANSIENT`` durability."

Reply:
Please add something to the section describing the tests about trying TRANSIENT_LOCAL and why we chose not to adopt it. On paper it looks better than TRANSIENT (less load on the daemons). So that information would be a useful lesson learned. I'd hate to have people waste time trying it again if we already know it won't work.